app/frontend/src/components/Scroll/ChevronScroll.tsx

```diff
@@ -0,0 +1,86 @@
+import React, { useRef, useCallback } from "react";
+import { IconChevronLeft, IconChevronRight } from "assets/images/images";
+import { combineClasses } from "components/Utility/utils";
+
+function ChevronScroll(props: React.PropsWithChildren<{}>) {
+  const [showRightChevron, setShowRightChevron] = React.useState(true);
+  const [showLeftChevron, setShowLeftChevron] = React.useState(false);
+  const [childIndex, setChildIndex] = React.useState(0);
+
+  const scrollRef = useRef<HTMLDivElement>(null);
+
+  const handleChevronVisibility = useCallback(
+    (e: React.UIEvent<HTMLDivElement>) => {
+      const { scrollLeft, scrollWidth, clientWidth } = e.currentTarget;
+      const maxScroll = scrollWidth - clientWidth - 1;
+      // show left chevron if not at start of scrollable area
+      setShowLeftChevron(scrollLeft !== 0);
+      //show right chevron if not at end of scrollable area
+      setShowRightChevron(scrollLeft < maxScroll);
+    },
+    []
+  );
+
+  const scrollMove = (direction: string) => {
+    if (scrollRef.current) {
+      const { children } = scrollRef.current;
+      // Move the scroll bar left or right by the width of each child element
+      if (direction === "right") {
+        scrollRef.current.scrollBy({
+          left: children[childIndex].scrollWidth,
+          behavior: "smooth",
```

Comment:
Out of curiosity, did scroll snap not work at all? From the comment, it sounds like the purpose of this is to discern the width of the children items?

Reply:
Ya so we are using scroll snap but in order to snap from the start of one child element to the next you have to move more than half the distance of the child element. I'm open to suggestions for a better way to do this, but have been unable to discover a viable alternative thus far


[screen-capture (2).webm](https://github.com/hackforla/CivicTechJobs/assets/73561520/c95fbf71-743a-47f2-915d-ebbbfecff05a)

Reply:
Ahh, I see. That is pretty rough. Thanks for the explanation!